iothub/device/src/IotHubDeviceClient.cs

```diff
@@ -109,6 +109,22 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         public void SetConnectionStateChangeHandler(Action<ConnectionState, ConnectionStateChangeReason> stateChangeHandler)
             => InternalClient.SetConnectionStateChangeHandler(stateChangeHandler);
 
+        /// <summary>
+        /// Last connection state.
+        /// </summary>
+        public ConnectionState LastConnectionState
+        {
+            get => InternalClient._lastConnectionState;
+        }
```

Comment:
Btw, you can write it this way in just one line.

```suggestion
        public ConnectionState LastConnectionState => InternalClient._lastConnectionState;
```